bootstrap/bootstrap.go

```diff
@@ -672,6 +672,13 @@ func (b *Bootstrap) executePluginHook(name string, checkouts []*pluginCheckout)
 	hookTypeSeen := make(map[string]bool)
 
 	for i, p := range checkouts {
+		// First we verify the plugin actually implements the hook
+		hookPath, err := b.findHookFile(p.HooksDir, name)
+		if err != nil {
```

Comment:
I think we should check for the `os.ErrNotExist` that `findHookFile` returns, rather than assuming _any_ error is that scenario. That'll stop this same issue from being triggered by some future possible error condition other than the hook file not existing.

Reply:
Yep, good idea. So perhaps:

```go
		// First we verify the plugin actually implements the hook
		hookPath, err := b.findHookFile(p.HooksDir, name)
		if err == os.ErrNotExist {
			continue
		} else if err != nil {
			return err
		}
```

Reply:
I was worried about the future compatibility of that; if `findHookFile` wraps the error then that equality check will no longer be true. I think this will do the trick: https://godoc.org/os#IsNotExist

Pushed as d2f3c9056531426553cd13669750533ce313d070

Reply:
Unfortunately `os.IsNotExist` doesn't actually handle unwrapping:
https://play.golang.org/p/AaGYo-6pFoV

```
	// Note that this function is not errors.Is:
	// underlyingError only unwraps the specific error-wrapping types
	// that it historically did, not all errors.Wrapper implementations.
```
-- https://golang.org/src/os/error.go?s=2847:2878#L82

I guess there's hope that in future it will do unwrapping, though?
And until then, we're only returning that one error type directly, so it'll probably be fine?

Using proper `errors.Is()` unwrapping here would make the agent incompatible with Go prior to v1.13, which I think we'd want to avoid for a bit longer.

Reply:
I pushed some code comments to keep track of this.